src/NUnitEngine/nunit.engine/Services/TestAgency.cs

```diff
@@ -200,7 +202,7 @@ private Guid LaunchAgentProcess(TestPackage package)
                 case RuntimeType.Mono:
                     p.StartInfo.FileName = NUnitConfiguration.MonoExePath;
                     string monoOptions = "--runtime=v" + targetRuntime.ClrVersion.ToString(3);
-                    if (enableDebug) monoOptions += " --debug";
+                    if (debugTests || debugAgent) monoOptions += " --debug";
```

Comment:
According to the docs:

--debug
             Turns on the debugging mode in the runtime.  If an assembly  was
             compiled with debugging information, it will produce line number
             information for stack traces.

We have been setting this for a long time and it doesn't work on every platform but appears to do no harm. For this change, I figured it should be used for DebugAgent as well as DebugTests.